src/library/vm/vm.cpp

```diff
@@ -317,6 +318,35 @@ void vm_obj_cell::dealloc() {
     }
 }
 
+unsigned hash(vm_obj const & o) {
```

Comment:
```suggestion
unsigned hash(vm_obj const & o) {
    check_system("hash(vm_obj)");
```

This will at least output an error message on stack overflow.